lib/chef/knife/vsphere_vm_state.rb

```diff
@@ -110,6 +110,9 @@ def run
       when 'reset'
         vm.ResetVM_Task.wait_for_completion
         puts "Reset virtual machine #{vmname}"
+      when 'reboot'
+        vm.RebootGuest
```

Comment:
Aha, thanks for that explanation. Also, thanks for putting this PR together! Could I trouble you to squash it down to one commit and I'll merge it?

Thanks.